container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go

```diff
@@ -1095,9 +1120,14 @@ func (manager *DockerManager) isImageAvailableLocally(ctx context.Context, image
 	return numMatchingImages > 0, nil
 }
 
-func (manager *DockerManager) pullImage(context context.Context, imageName string) (err error) {
+func (manager *DockerManager) pullImage(context context.Context, imageName string) error {
+	// As we're using the docker client with no timeout to pull the image, we quickly check with the client that has
+	// a timeout whether the docker engine is reachable.
+	if _, err := manager.dockerClient.Ping(context); err != nil {
```

Comment:
not important to do now, but I wonder how we could force this call always before any call with the Docker client without a timeout, a wrapper maybe? but not important to do now at least you have a quick and easy solution

Reply:
Yes that would be nice, though I can't think a good simple implementation for such a wrapper, b/c docker Client object is a struct, not an interface. We could do it at the docker_manager level though. Maybe if at some point we end up with a lot of endpoints using the NoTimeout client, for now as it is only two maybe it's fine to keep it simple